sole migration updates

mle2718 opened the pull request.

mle2718 commented:
@jbwalden and @SWerner2  -- I've mostly updated the metadata to account for the sole migration. Could 1 (or both) of you review the changes (Files changed tab on top or the green "review" button).  This repo is set to public, and the only way I could think to have a moderate amount of quality control on it is to require a review before anything can be committed to the main branch.

SWerner2 commented on AMS_DAS.md:
Does this mean archived?

SWerner2 commented on AMS_DAS.md:
```diff
@@ -1,11 +1,12 @@
-Location: Sole
+AMS can be found in the NEFSC_GARFO schema. The relevant tables are prefixed with AMS_.
+DAS can be found in the NEFSC_GARFO schema, the relevant tables are prefixed with DAS_.
+DAS2 have been removed from SOLE and cannot be accessed through the GARFO_NEFSC dblink.  
```
would suggest changing to archived and potentially reference cold storage.

mle2718 replied:
[725b86b](https://github.com/NEFSC/READ-SSB-Lee-metadata/pull/27/commits/725b86be5c2f1f6e5c8e6b812d59ed563fda9ac1)

SWerner2 commented on Client.md:
```diff
@@ -1,14 +1,14 @@
 # Overview
 
-All about the Client Schema
+All about the Client schema.
```
Little confusing because it is not its own schema anymore it is under NEFSC_GARFO.

SWerner2 commented on DMIS.md:
```diff
@@ -2,20 +2,20 @@
-* APSD.t_ssb_trip_current@garfo_nefsc
-* APSD.t_ssb_catch_current@garfo_nefsc 
-* APSD.t_ssb_discard_current@garfo_nefsc
+* NEFSC_GARFO.APSD_t_ssb_trip_current@garfo_nefsc
```
Are these still accessible through db links?

These have been added to the nefsc_garfo pipeline db links on sole wont work in the future.

mle2718 replied:
[0cb6924](https://github.com/NEFSC/READ-SSB-Lee-metadata/pull/27/commits/0cb69246e59d3ff08c8211848e0c3cc1489105f5) and [0cb6924](https://github.com/NEFSC/READ-SSB-Lee-metadata/pull/27/commits/0cb69246e59d3ff08c8211848e0c3cc1489105f5)

SWerner2 commented on DMIS.md:
````diff
@@ -61,7 +61,7 @@ As of Feb 2022, I'm not sure if (1) is still true:
 ```
 select docid_length,count(*) from(
-select length(to_char(docid)) as docid_length from APSD.t_ssb_trip_current@garfo_nefsc where docid is not null) group by docid_length;
+select length(to_char(docid)) as docid_length from NEFSC_GARFO.APSD_t_ssb_trip_current@garfo_nefsc where docid is not null) group by docid_length;
````
again the @garfo_nefsc tag is not in the name of the table on NEFSC_GARFO

mle2718 replied:
0cb6924 and 0cb6924

SWerner2 commented on DMIS.md:
````diff
@@ -196,7 +196,7 @@ For many fisheries in the Greater Atlantic Region, DMIS (APSD.t_ssb_catch_curren
 ```
 select XYZ
-from APSD.t_ssb_trip_current@garfo_nefsc T, APSD.t_ssb_catch_current@garfo_nefsc C
+from NEFSC_GARFO.APSD_t_ssb_trip_current@garfo_nefsc T, NEFSC_GARFO.APSD_t_ssb_catch_current@garfo_nefsc C
````
same as above.

SWerner2 commented on MQRS.md:
```diff
@@ -1,11 +1,13 @@
 
 The MQRS system is fun. It contains information on "eligibilities."  The sector rosters are also stored here.  It is used in conjunction with other tables, but not cross-referenced against them for logical consistency.
+
+The SECTOR schema still exists on SOLE.
```
Sole is going away so what does this mean?

mle2718 replied:
Sole is still there (as of today), so I don't want to delete this line right now.

mle2718 replied:
issue #31. We'll pull it when sole is turned off.

SWerner2 commented on README.md:
````diff
@@ -35,18 +35,14 @@ This has consequences for reproducibility if you do not store a copy of the data
 1.  It's good practice to include the schema and server when you query data. That is, write:
 ```
-select * from permit.vps_owner@sole
+select * from nefsc_garfo.permit_vps_owner@NEFSC_USERS
````
are the @ symbols needed?

mle2718 replied:
They are not. It's explicit about the location of the database, which is good practice that nobody follows.

mle2718 replied:
> Do you note the difference between the description of NEFSCDB1 vs NEFSC_USERS? Potentially making a Server overview md file?

This (NEFSCDB1 vs NEFSC_USERS) depends on your local configuration. On my laptop, ITD set up the connection name "NEFSC_USERS" to point to NEFSCDB1.  On the SAS container, they renamed the "nova" connection name to point to NEFSCDB1.  I will tidy up this text

schema and server 

to schema and connection name.

I think a Server overview file would be a great idea. It's beyond the scope of this PR.

SWerner2 commented on code_fragments/DMIS_observer_match.sas:
```diff
@@ -3,15 +3,15 @@
 
 proc sql;
-	connect to oracle (user = gardini password = "XXXXXXXXXXXX" path = sole);
+	connect to oracle (user = gardini password = "XXXXXXXXXXXX" path = nova);
```
nova?

mle2718 replied:
nova is the connection string that currently works on the SAS container.

SWerner2 commented on permit.md:
```diff
@@ -1,19 +1,16 @@
 
 
-*the BUS_OWN table is not the main view for business onwer information. THat is found here: Client.bus_own@garfo_nefsc. For more information on this system see Client.md
+The following views can be found in the NEFSC_GARFO schema on nefscdb1
```
AMS_ALLOCATION_TX
AMS_DAS_ACTIVITY_CODES
AMS_LEASE_EXCH_APPLIC
AMS_OBSERVER_TRIP
AMS_TRIP
AMS_TRIP_AND_CHARGE
AMS_TRIP_LANDING
APSD_T_SSB_CATCH_CURRENT
APSD_T_SSB_DISCARD_CURRENT
APSD_T_SSB_TRIP_CURRENT
CATCH
CFDBS_AREA
CFDBS_LOC
CFDBS_LOC2AREAS
CFDBS_SPECIES_ITIS_NE
CFDBS_STATENM
CFDBS_VALID_PORTS
CFDERS_ALL_YEARS
CLIENT_BUS_OWN
DOCUMENT
IMAGES
JOPS_OPERATOR
MQRS_MORT_ELIG_CRITERIA
PERMIT_DEALER
PERMIT_OWNER_HISTORY_PIDS
PERMIT_VALID_FISHERY
PERMIT_VPS_FISHERY_NER
PERMIT_VPS_OWNER
PERMIT_VPS_VESSEL
SCBI_SPECIES_ITIS_NE
SECTOR_ACE_MRI_AS_POUNDS
SECTOR_MRI_PERMIT_VESSEL_HISTORY
SECTOR_PSC_MRI_AS_PERCENT
SECTOR_TRANSFERS
SECTOR_TRANSFER_STOCK
SFCLAM_SFOQVR
TRIP_REPORTS_CATCH
TRIP_REPORTS_DOCUMENT
TRIP_REPORTS_IMAGES
VTR_DEALR
VTR_VLGEAR
VTR_VLSPPTBL

mle2718 replied:
I don't want to put this information into the permit.md file.

mle2718 replied:
This is going to be a giant pain in the ass if we want to keep our metadata organized by schema. -- because then everything should go into a single "GARFO_NEFSC" file. I don't want to do that, because it's just gross.

mle2718 replied:
@SWerner2  -- okay, I have created an issue based on this comment.

SWerner2 commented on veslog.md:
```diff
@@ -1,5 +1,5 @@
 # Warning
-As of at least January 6, 2023 this table is no longer being supported by ITD. This means the table is not being updated and users should switch to GARFO VTR tables and ultimately CAMS when available.
+As of at least January 6, 2023 this table is no longer being supported by ITD. This means the table is not being updated and users should switch to GARFO VTR tables (Catch, Images, and Document tables in the NEFSC_GARFO schema) and ultimately CAMS when available.
```
CATCH IMAGES and  DOCUMENT this may change to TRIP_REPORTS_CATCH
TRIP_REPORTS_DOCUMENT
TRIP_REPORTS_IMAGES but I am checking in on it

mle2718 replied:
They are both there. I don't know why.

mle2718 replied:
[bdc4063](https://github.com/NEFSC/READ-SSB-Lee-metadata/pull/27/commits/bdc4063b34c903c7fe46bd65c26132b660e5fa1f)

mle2718 replied:
I'm going to resolve this, but open an issue for later. Both of these are in the NEFSC_GARFO schema.

mle2718 replied:
added to #30. we'll sort it out when ITD does.

SWerner2 commented on veslog.md:
```diff
@@ -153,16 +191,13 @@ If you need to extract many years of VESLOG data, do it in a loop.
-  from noaa.document@garfo_nefsc d,  -- vessel permit, sail, land, crew, trip category (commercial, P/C, rec)
-       noaa.images@garfo_nefsc i,    -- Subtrip information, gear, effort, area
-       noaa.catch@garfo_nefsc c      -- Species, kept, discarded, who sold to, ddate_sold
+  from NEFSC_GARFO.TRIP_REPORTS_DOCUMENT d,  -- vessel permit, sail, land, crew, trip category (commercial, P/C, rec)
```
This information conflicts with what is listed above, what is true souce CATCH or TRIP_REPORTS_CATCH

mle2718 replied:
[bdc4063](https://github.com/NEFSC/READ-SSB-Lee-metadata/pull/27/commits/bdc4063b34c903c7fe46bd65c26132b660e5fa1f)

mle2718 replied:
Added to #30. We'll update it when ITD sorts it out.

SWerner2 replied:
Would suggest keeping this open or adding a note so it is not forgotten then we are suggesting a source which is not verified

SWerner2 commented:
@mle2718 added comments for your review.

mle2718 commented:
> @mle2718 added comments for your review.

@SWerner2 --thanks. I've fixed most and moved 2 to issues. not sure what to do with the gigantic NEFSC_GARFO schema.

jbwalden reviewed (commented):
CAMS has replaced DMIS. Do we need to include DMIS in the updates?

mle2718 commented:
@jbwalden :

> CAMS has replaced DMIS. Do we need to include DMIS in the updates?

I think so, people may find some thing here useful as they move off DMIS.  

I'm waiting on GARFO to get back to me with the end date of DMIS.  Currently, it has data through April 30, 2023, but the final year is not "complete." When that comes in, I'll make a couple last changes

jbwalden commented:
I still don't agree about DMIS, but it is your call. CAMS started from DMIS, but is different in that it is more collaborative between the NEFSC and GARFO, and the QC is better. I think people need to make the transition to CAMS and forget about DMIS. Hanging onto DMIS doesn't help us move forward, in my opinion.

mle2718 commented:
> I still don't agree about DMIS, but it is your call. CAMS started from DMIS, but is different in that it is more collaborative between the NEFSC and GARFO, and the QC is better. I think people need to make the transition to CAMS and forget about DMIS. Hanging onto DMIS doesn't help us move forward, in my opinion.

@jbwalden  -- As usual, we agree far more than we disagree. I've added the end date for DMIS to the metadata in f68e8f488bef48b22e8781730b1943867f63ea3f.  I don't think it's appropriate to wipe DMIS from our collective memory, but I think it's pretty clear that DMIS has been declared obsolete in favor of CAMS.

jbwalden commented:
Min-Yang, I am having a hard time following everything in Github. It seems like you are asking me to review something specifically, but I can't follow which thread it is (or topic). Will try to look again tomorrow with a fresh set of eyes.

mle2718 commented:
> Min-Yang, I am having a hard time following everything in Github. It seems like you are asking me to review something specifically, but I can't follow which thread it is (or topic). Will try to look again tomorrow with a fresh set of eyes.

@jbwalden  -- Oh sorry about that.  If you go up to the top and find the "Files Changed" tab, you'll see a split-screen (original on the left and revised version on the right).  You can leave specific comments on individual lines by hovering over the + or - next to the line numbers and clicking the blue squares.  If you do that, the "suggestion" button (the +-) is really useful.

If it looks good, you can use the green "Review Changes" tab to approve. If it looks bad, you can comment or request changes.

mle2718 commented:
Thanks @jbwalden .
@SWerner2 -- are you okay with this, or are there more changes that you'd like to see?

SWerner2 commented on svdbs.md:
```diff
@@ -1,9 +1,13 @@
 
 Most of the information there has been processed to some degree.
 
+For access, email paul.kostovick@noaa.gov
```
Recommend linking the whole app schema inventory somewhere: https://docs.google.com/spreadsheets/d/15FtGnNUgct7mTsRpPP9xX4BLkceY7SfMnZvdA_kjlxY/edit#gid=1754518543

mle2718 replied:
It's in the main readme.

jbwalden replied:
Does anyone in the branch use svdbs?

mle2718 replied:
> Does anyone in the branch use svdbs?

I use it to pull data for age-length keys for the Groundfish Bioeconomic Rec model. I'm guessing @acarrharris does as well for Scup, SF, and BSB.

mle2718 replied:
they are.

SWerner2 commented on veslog.md:
The veslog series, (including veslogxxxxt veslogxxxxg veslogxxxxs)

mle2718 replied:
@SWerner2  -- I can't figure out what this comment is attached to.

SWerner2 commented on veslog.md:
Where is this recommendation coming from? Is this center guidance or best practice by the branch?

mle2718 replied:
@SWerner2  -- I can't figure out what this comment is attached to.

SWerner2 commented on veslog.md:
Would suggest adding code for folks to covert veslog to CATCH IMAGES and DOCUMENT

SWerner2 reviewed (commented):
@mle2718 Additional comments added.

SWerner2 commented:
> Thanks @jbwalden . @SWerner2 -- are you okay with this, or are there more changes that you'd like to see?

@mle2718 please see additional comments. Thank you.

mle2718 commented on MQRS.md:
````diff
@@ -73,7 +48,7 @@ And stitching results together with a union.
   
   ```
-select * from mqrs.mort_elig_criteria
+select * from NEFSC_GARFO.mqrs_mort_elig_criteria
````
I've added this: 
At GARFO, MQRS.MORT_ELIG_CRITERIA is not archived and is a current production table. I strongly suggest using a view to interpret it, because there are some pitfalls in the table structure that aren't obvious.(torey.adler on 15/Dec/23 11:45 AM)

although it's mildly infuriating that all we get are "there are some pitfalls"

SWerner2 replied:
This is helpful! Thank you. And yes, but I think there are going to be efforts to communicate more thoroughly with GARFO about authoritative data sources where we can address these pitfalls. So, there is that!
